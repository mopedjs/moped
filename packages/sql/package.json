{
  "name": "@moped/sql",
  "version": "0.0.1",
  "description": "",
  "main": "./lib/index.js",
  "types": "./lib/index.d.ts",
  "dependencies": {
    "@types/node": "^8.0.33",
    "pg-sql": "^1.1.0"
  },
  "scripts": {
    "prepublish": "tsc -p tsconfig.build.json"
  },
  "repository": "https://github.com/mopedjs/moped/tree/master/packages/sql",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  }
}
