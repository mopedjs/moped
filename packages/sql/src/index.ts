import {sql, SQLQuery} from 'pg-sql';

export {SQLQuery};

/**
 * The interface we actually expect people to use.
 */
export interface SQL {
  (strings: TemplateStringsArray, ...values: Array<any>): SQLQuery;

  join(queries: Array<SQLQuery>, seperator?: string): SQLQuery;
  __dangerous__rawValue(text: string): SQLQuery;
  value(value: any): SQLQuery;
  ident(...names: Array<any>): SQLQuery;
}

// Create the SQL interface we export.
const modifiedSQL: SQL = Object.assign(
  (strings: TemplateStringsArray, ...values: Array<any>): SQLQuery =>
    sql(strings, ...values),
  {
    join: sql.join,
    __dangerous__rawValue: sql.raw,
    value: sql.value,
    ident: sql.ident,
  },
);

export default modifiedSQL;
