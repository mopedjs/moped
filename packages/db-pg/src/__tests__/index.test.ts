import createDb from '@moped/db-pg-create';
import connect, {Connection} from '../';
import sql from '@moped/sql';

process.env.DATABASE_URL = 'postgres://moped-db-pg@localhost/moped-db-pg';

let db: Connection = null as any;

test('create database', async () => {
  await createDb();
  db = connect();
});

test('error messages', async () => {
  try {
    await db.query(sql`
      SELECT * FROM foo;
      SELECT * FROM bar WHERE id = ${'Hello World, Goodbye World, etc.'};
      SELECT * FRM baz;
      SELECT * FROM bing;
    `);
  } catch (ex) {
    expect(ex.message).toMatchSnapshot();
    return;
  }
  expect(false).toBe(true);
});
