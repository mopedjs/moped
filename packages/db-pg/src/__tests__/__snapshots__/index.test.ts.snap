// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error messages 1`] = `
"syntax error at or near \\"FRM\\"

  2 |       SELECT * FROM foo;
  3 |       SELECT * FROM bar WHERE id = $1;
> 4 |       SELECT * FRM baz;
    |                ^^^
  5 |       SELECT * FROM bing;
  6 |     
"
`;
